### PR TITLE
ISPN-12522 + ISPN-12578 Upgrade to Hibernate Search 6.0.0.CR2 then 6.0.0.Final

### PR DESCRIPTION
**Pull request by yrodiere**
https://issues.redhat.com/browse/ISPN-12522
https://issues.redhat.com/browse/ISPN-12578

The first commit is the actual upgrade, the following commits just take advantage of changes in recent versions of Hibernate Search (including CR2) to simplify the code or make it more consistent.

The final commit upgrades to 6.0.0.Final, which is identical for our purposes.

### CODE REVIEW

File: query/src/main/java/org/infinispan/query/stats/impl/LocalIndexStatistics.java [hunk 1]
```diff
@@ -36,7 +37,8 @@
                Class<?> javaClass = e.javaClass();
                SearchScope<?> scope = session.scope(javaClass, e.name());
                long count = session.search(scope).where(SearchPredicateFactory::matchAll).fetchTotalHitCount();
-               return new IndexInfo(count, 0);
+               long size = e.indexManager().unwrap(LuceneIndexManager.class).computeSizeInBytes();
+               return new IndexInfo(count, size);
```
Comment (gustavonalle) [inline query/src/main/java/org/infinispan/query/stats/impl/LocalIndexStatistics.java:41]: Thanks @yrodiere , this was not forgotten really, I was planning to send a follow up PR to decouple it from the upgrade itself, but you saved me some time  :+1:

**Comment by gustavonalle** [comment]
Changes LGTM. Triggered a CI run as the others didn't complete

**Comment by yrodiere** [comment]
I rebased and force-pushed.

Regarding failures, as far as I can tell only this one was caused by my changes: https://ci.infinispan.org/job/Infinispan/job/PR-8894/3/testReport/junit/org.infinispan.rest.search/EmbeddedRestSearchTest/Blocking/

BlockHound is complaining about a blocking call. But the method that I changed was already relying on a blocking call: `session.search(scope).where(SearchPredicateFactory::matchAll).fetchTotalHitCount()`. Do we need this code to be non-blocking? If so, the current signature, `Map<String, IndexInfo> indexInfos()`, is definitely not adapted.

FWIW, there's a method to compute index size asynchronously, so we could do `e.indexManager().unwrap(LuceneIndexManager.class).computeSizeInBytesAsync();`. The only problem here is the method signature: we would have to return a CompletionStage<Map<String, IndexInfos>>`, not a `Map<String, IndexInfos>`.

**Comment by gustavonalle** [comment]
Changing to  ```CompletionStage<Map<String, IndexInfos>>``` would not be an issue. This is a new interface introduced a month ago, not released yet

**Comment by gustavonalle** [comment]
> But the method that I changed was already relying on a blocking call: session.search(scope).where(SearchPredicateFactory::matchAll).fetchTotalHitCount()

It could be BlockHound did not catch it blocking in the test suite

**Comment by yrodiere** [comment]
Probably. Ok, I'll have a look soon if you don't.

**Comment by gustavonalle** [comment]
No worries @yrodiere, feel free to ignore that test I can fix it together with other failures

**Comment by yrodiere** [comment]
Hi @gustavonalle! I rebased on master and added a commit to upgrade to 6.0.0.Final. Any chance this gets into the CR1? Do you need help?

**Comment by anistor** [comment]
Hi @yrodiere . I'm looking at this while @gustavonalle is away. So far it looks good, but I have to admit I'm way behind with catching up on the changes in this area. I'll merge after I have some more CI runs

**Comment by yrodiere** [comment]
Thanks @anistor .

There was just one problem that @gustavonalle wanted to fix (see previous comments). The problem was probably pre-existing; basically the collection of stats is done in a thread that should not execute blocking code, but the methods are not defined with reactive in mind, so we can only do synchronous, blocking calls. I suppose a quick fix would be to run stat collection in a worker thread?

**Comment by anistor** [comment]
I wonder if rebasing on top of latest master would solve some of the CacheV2ResourceTest failures. I hope it will. Could you please rebase @yrodiere ?

**Comment by yrodiere** [comment]
@anistor Done. Let's see...

**Comment by anistor** [comment]
Seems like the CacheV2ResourceTest.testSearchStatistics failure is legit. The index stats after the clearSearchStats operation are incorrect. The count (number of indexed entities) is 0 but the size in bytes of the index is still 69. 

The other failures seem to happen randomly on CI but I do not see them locally.

**Comment by yrodiere** [comment]
Ok, I'm on it... I'll convert the code to async/reactive as well, to avoid the other failure in `CacheV2ResourceTest`.

**Comment by yrodiere** [comment]
@anistor It should work now. Blockhound seems happy when I run tests locally.

**Comment by anistor** [comment]
Almost there! I'll try to fix org.infinispan.server.functional.HotRodCacheQueries failures and merge

**Comment by yrodiere** [comment]
@anistor FYI the failures on `HotRodCacheQueries` are not directly related to this PR, they are also present on master: https://ci.infinispan.org/job/Infinispan/job/master/2513/

**Comment by anistor** [comment]
Merged in master. Thanks @yrodiere !

**Pull request closed without merging**